### ISSUE TITLE
Hitbox disappearance issue after adding animation with sprite_add function runtime 2024.200.0.502 and 2023.11.1.160

### ISSUE THREAD

**Issue by yevhenii-sir** [YoYoGames/GameMaker-Bugs#4712]
### Description

After updating the runtime from version 2.3.5.458 to version 2023.11.1.160, hitboxes/bounding boxes disappear when loading Spine animations through the sprite_add function. Before this, I converted the animation from Spine v3.5.43 to v4.0.64.

After adding a sprite using the sprite_add function, the hitboxes/bounding boxes in the animation disappear. 
**However, when I add the animation to the project assets, everything works correctly.**

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/36339434/a4cb9da4-4442-426d-9126-065aa0a5f2be)

Example project: [Collision_bug.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14270042/Collision_bug.zip)
Contact Us Package: [Package.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14270106/Package.zip)

### Expected Change

To ensure that the hitbox does not disappear when loading Spine animation via sprite_add, so everything works as when adding animation to assets.

### Steps To Reproduce

1. Start GameMaker
2. Add animation in datafiles
3. Load animation with sprite_add
4. Draw bbox
5. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.11.1 (Monthly), 2024.200 (Betas)

### Which platform(s) are you seeing the problem on?

Windows, Android

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by yevhenii-sir** [comment]
@yoyo-danielc can you help me?

**Comment by yevhenii-sir** [comment]
The solution might be adding automatic collision mask type assignment in "Spine Collision Mesh" for loaded JSON animations via sprite_add.

**Comment by yoyo-danielc** [comment]
Fixes are merged into the 2026.06 branch now, should be in the next beta.

Bounding boxes of Spine sprites loaded at runtime are now initialised to "Full image" which covers the full area covered by the sprite's setup pose, and closely matches the default behavior when importing a sprite in the IDE.

You can switch to using Spine collision meshes at runtime by using `sprite_collision_mask()`:

```
sprite_collision_mask(sprite_index, false, bboxmode_fullimage, 0, 0, 0, 0, bboxkind_spine, 0);
```

Or back to the default full image bounding box:

```
sprite_collision_mask(sprite_index, false, bboxmode_fullimage, 0, 0, 0, 0, bboxkind_rectangular, 0);
```

Or to a manual bounding box:

```
sprite_collision_mask(sprite_index, false, bboxmode_manual,
		<left>, <top>, <right>, <bottom>,
		bboxkind_rectangular, 0);
```

I've updated your test project to demo this functionality.

[Collision_bug.zip](https://github.com/user-attachments/files/15546569/Collision_bug.zip)

QA Note: I had to refactor a good chunk of the bounding box handling in the HTML5 runner, not only for Spine sprites but also Bitmap/SWF sprites.

Doc Note: The documentation for `sprite_collision_mask()` needs updating with the new `bboxkind_spine` constant, and it will now work on Spine sprites, but only for the specific modes I've documented above.

**Comment by mgeddesGM** [comment]
verified as of Beta IDE v2024.600.0.579  Runtime v2024.600.0.605
